### PR TITLE
Allow user to specify the center of map coordinates in `pfsspy.utils.carr_cea_wcs_header`

### PR DESCRIPTION
**Pull request by STBadman**
Currently, the function `pfsspy.utils.carr_cea_wcs_header` assumes that it is being constructed for a Carrington map where the center is at 0 degrees Carrington longitude. This change adds a keyword `map_center_longitude` which must be passed with `astropy.units.deg` units. The default value is 0 meaning the default behavior of the function is unchanged but now a user can manually change it, for example using 180 degrees to align Carrington 0 with the left hand edge of the map.

Although its uncommon to not be loading a magnetogram directly from a FITS file in which case the header is created automatically with the correct roll, I had a use a case where I needed this functionality: I had saved a CEA magnetogram data as a `numpy.array` and wanted to reconstruct a `sunpy.map.Map` from it, however the `np.array` data did have CR0 at the left hand edge and so the default behavior of the header creation was incorrect and led to a misaligned map. This fix allows the user to control this.

The build of the PR has failed for py38 due to some matplotlib deprecation warnings around get_cmap. This looks tied up in sunpy and I don't see an easy fix - can this PR still be merged?

### CODE REVIEW

File: pfsspy/utils.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def load_adapt(adapt_path):
     return adaptMapSequence
 
 
-def carr_cea_wcs_header(dtime, shape):
+def carr_cea_wcs_header(dtime, shape, map_center_longitude=0*u.deg):
```
Comment (dstansby) [inline pfsspy/utils.py:91]:
```suggestion
def carr_cea_wcs_header(dtime, shape, *, map_center_longitude=0*u.deg):
```
I think making the argument keyword only makes sense here.

File: pfsspy/utils.py [hunk 2]
```diff
@@ -100,18 +100,32 @@ def carr_cea_wcs_header(dtime, shape):
     shape : tuple
         Map shape. The first entry should be number of points in longitude, the
         second in latitude.
+    map_center_longitude : astropy.Quantity
```
Comment (dstansby) [inline pfsspy/utils.py:103]:
```suggestion
    map_center_longitude : astropy.units.Quantity
```

File: pfsspy/utils.py [hunk 3]
```diff
@@ -100,18 +100,32 @@ def carr_cea_wcs_header(dtime, shape):
...
     """
+    # Check map_center_longitude has units of degrees
+    assert hasattr(map_center_longitude, "unit"), (
+        f"Keyword `map_center_longitude`={map_center_longitude} has no "
+        + f"`astropy.units` attribute, must have units and be `astropy.units.deg`."
+    )
+    assert map_center_longitude.unit == u.deg, (
+        f"Keyword `map_center_longitude`={map_center_longitude} does not have "
+        +f"units of `astropy.units.deg`"
+    )
```
Comment (dstansby) [inline pfsspy/utils.py:121]: Instead of doing these checks manually, you can use https://docs.astropy.org/en/stable/api/astropy.units.quantity_input.html

**Review by dstansby — CHANGES_REQUESTED** [review]
Looks 👍 - left some suggestions. Could you add a quick test to make sure this works in the future too?

File: pfsspy/utils.py (hunk 3 above)
Comment (STBadman) [reply pfsspy/utils.py]: Change made with https://github.com/dstansby/pfsspy/pull/358/commits/2e015ab6acc7326ba7ac8a9ca638d766b4ed54a7

**Comment by STBadman** [comment]
Thanks! Accepted your changes, and added a test for the cea header function. It checks that the default and user specified reference pixel longitudes are as expected and it checks the error handling of the keyword units is correct.

File: pfsspy/tests/test_utils.py [hunk 4]
```diff
@@ -115,3 +117,36 @@ def test_roll_map(gong_map):
     with pytest.raises(ValueError,
                        match='lh_edge_lon must be in'):
         utils.roll_map(adapt_map, lh_edge_lon=361 * u.deg)
+
+
+def test_cea_header():
+    # Assert default reference pixel is at 180 deg lon
```
Comment (dstansby) [inline pfsspy/tests/test_utils.py:123]:
```suggestion
    # Assert default reference pixel is at 0 deg lon
```

**Review by dstansby — APPROVED** [review]

**Pull request merged**